src/robot/result/model.py

```diff
@@ -740,6 +744,14 @@ def elapsedtime(self):
         return sum(child.elapsedtime for child in
                    chain(self.suites, self.tests, (self.setup, self.teardown)))
 
+    @setter
+    def suites(self, suites: 'Sequence[TestSuite|DataDict]') -> 'TestSuites[TestSuite]':
+        return TestSuites['TestSuite'](self.__class__, self, suites)
+
+    @setter
+    def tests(self, tests: 'Sequence[TestCase|DataDict]') -> TestCases['TestCase']:
```

Comment:
- Quotes around `TestCase` in `-> TestCases['TestCase']` probably aren't needed.
- I think `-> 'TestSuites[TestSuite]'` could also be `-> TestSuites['TestSuite']`. I'm not certain which is better, though.